src/realm/integer_compressor.cpp

```diff
@@ -265,14 +206,105 @@ bool IntegerCompressor::init(const char* h)
+{
+    return PackedCompressor::get_all(arr.m_integer_compressor, begin, end);
+}
+
+std::vector<int64_t> IntegerCompressor::get_all_flex(const Array& arr, size_t begin, size_t end)
+{
+    return FlexCompressor::get_all(arr.m_integer_compressor, begin, end);
+}
+
+void IntegerCompressor::get_chunk_packed(const Array& arr, size_t ndx, int64_t res[8])
+{
+    PackedCompressor::get_chunk(arr.m_integer_compressor, ndx, res);
+}
+
+void IntegerCompressor::get_chunk_flex(const Array& arr, size_t ndx, int64_t res[8])
+{
+    FlexCompressor::get_chunk(arr.m_integer_compressor, ndx, res);
+}
+
+void IntegerCompressor::set_packed(Array& arr, size_t ndx, int64_t val)
+{
+    PackedCompressor::set_direct(arr.m_integer_compressor, ndx, val);
+}
+
+void IntegerCompressor::set_flex(Array& arr, size_t ndx, int64_t val)
+{
+    FlexCompressor::set_direct(arr.m_integer_compressor, ndx, val);
+}
+
+template <class Cond>
+bool IntegerCompressor::find_packed(const Array& arr, int64_t val, size_t begin, size_t end, size_t base_index,
+                                    QueryStateBase* st)
+{
+    return PackedCompressor::find_all<Cond>(arr, val, begin, end, base_index, st);
+}
+
+template <class Cond>
+bool IntegerCompressor::find_flex(const Array& arr, int64_t val, size_t begin, size_t end, size_t base_index,
+                                  QueryStateBase* st)
+{
+    return FlexCompressor::find_all<Cond>(arr, val, begin, end, base_index, st);
+}
+
+void IntegerCompressor::set_vtable(Array& arr)
+{
+    static const Array::VTable vtable_packed = {get_packed,
+                                                get_chunk_packed,
+                                                get_all_packed,
+                                                set_packed,
+                                                {
+                                                    find_packed<Equal>,
+                                                    find_packed<NotEqual>,
+                                                    find_packed<Greater>,
+                                                    find_packed<Less>,
+                                                }};
+    static const Array::VTable vtable_flex = {get_flex,
+                                              get_chunk_flex,
+                                              get_all_flex,
+                                              set_flex,
+                                              {
+                                                  find_flex<Equal>,
+                                                  find_flex<NotEqual>,
+                                                  find_flex<Greater>,
+                                                  find_flex<Less>,
+                                              }};
+    if (is_packed()) {
+        arr.m_vtable = &vtable_packed;
+    }
+    else {
+        arr.m_vtable = &vtable_flex;
+    }
+}
+
+int64_t IntegerCompressor::get(size_t ndx) const
```

Comment:
why don't follow the same patter as the others and have a getter for packed and a getter for flex?

Reply:
`CompressedDataFetcher` needs this.

Reply:
So this is only used by `CompressedDataFetcher`? The only 2 places where we use that is in lower/upper bound.